src/group_theory/pi_hom.lean

```diff
@@ -184,37 +184,8 @@ begin
     simp [hmem], }
 end
 
-lemma pow (k : ℕ) : (hom S f) ^ k = hom S (f ^ k) :=
-begin
-  change (to_fun ϕ f S) ^ k = to_fun ϕ (f ^ k) S,
-  induction S using finset.induction_on with i S hnmem ih,
-  { simp },
-  { simp only [ to_fun_insert_of_not_mem _ _ _ _ hnmem],
-    rw [(to_fun_commutes ϕ f f S i hnmem).mul_pow, ih, pi.pow_apply, map_pow] },
-end
-
-lemma hom_eq_one_of_eq_one (h : ∀ x ∈ S, f x = 1) : hom S f = 1 :=
-begin
-  change to_fun ϕ f S = 1,
-  induction S using finset.induction_on with i S hnmem ih,
-  { simp },
-  { simp only [ to_fun_insert_of_not_mem _ _ _ _ hnmem],
-    rw [h _ (finset.mem_insert_self _ _), ih (λ i h', h i (finset.mem_insert_of_mem h'))],
-    simp, },
-end
-
-lemma order_of_hom_dvd_prod_card [hfin : ∀ i, fintype (H i)]:
-  order_of (hom S f) ∣ (∏ i in S, fintype.card (H i)) :=
-begin
-  rw order_of_dvd_iff_pow_eq_one,
-  rw pow,
-  apply hom_eq_one_of_eq_one,
-  intros i hmem,
-  simp only [pi.pow_apply, pi.one_apply],
-  rw ← order_of_dvd_iff_pow_eq_one,
-  calc order_of (f i) ∣ fintype.card (H i)                   : order_of_dvd_card_univ
-                  ... ∣ (∏ (i : I) in S, fintype.card (H i)) : finset.dvd_prod_of_mem _ hmem,
-end
+lemma pow (k : ℕ) : (hom S f) ^ k = hom S (f ^ k) := by simp
+/- uses map_pow, and now unnecessary -/
```

Comment:
Silly me, for some reason I thought this doesn’t hold for all homomorphism. Thanks :-)